test/vars/MavenSpec.groovy

```diff
@@ -148,105 +151,123 @@ class MavenSpec extends JenkinsPipelineSpecification {
-            1 * getPipelineMock("sh")([script: 'mvn -B -N -e versions:update-child-modules -Dfull -DallowSnapshots=true -DgenerateBackupPoms=false', returnStdout: false])
+        1 * getPipelineMock("sh")([script: "mvn -B -N -e versions:update-parent -Dfull -DparentVersion=[${newVersion}] -DallowSnapshots=true -DgenerateBackupPoms=false", returnStdout: false])
+        1 * getPipelineMock("sh")([script: 'mvn -B -N -e versions:update-child-modules -Dfull -DallowSnapshots=true -DgenerateBackupPoms=false', returnStdout: false])
     }
 
     def "[maven.groovy] run mvn set version property"() {
         setup:
-            String newVersion = '1.2.3'
-            String propertyName = 'version.org.kie.kogito'
+        String newVersion = '1.2.3'
+        String propertyName = 'version.org.kie.kogito'
         when:
-            mavenGroovy.mvnSetVersionProperty(propertyName, newVersion)
+        mavenGroovy.mvnSetVersionProperty(propertyName, newVersion)
         then:
         1 * getPipelineMock("sh")([script: "mvn -B -e versions:set-property -Dproperty=$propertyName -DnewVersion=$newVersion -DallowSnapshots=true -DgenerateBackupPoms=false", returnStdout: false])
     }
 
     def "[maven.groovy] uploadLocalArtifacts"() {
         setup:
-            String mvnUploadCredsId = 'mvnUploadCredsId'
-            String mvnUploadCreds = 'user:password'
-            mavenGroovy.getBinding().setVariable('kieUnpack', mvnUploadCreds)
-            String artifactDir = '/tmp'
-            String repoUrl = 'https://redhat.com'
+        String mvnUploadCredsId = 'mvnUploadCredsId'
+        String mvnUploadCreds = 'user:password'
+        mavenGroovy.getBinding().setVariable('kieUnpack', mvnUploadCreds)
+        String artifactDir = '/tmp'
+        String repoUrl = 'https://redhat.com'
         when:
-            mavenGroovy.uploadLocalArtifacts(mvnUploadCredsId, artifactDir, repoUrl)
+        mavenGroovy.uploadLocalArtifacts(mvnUploadCredsId, artifactDir, repoUrl)
         then:
-            1 * getPipelineMock('usernameColonPassword.call')([credentialsId: mvnUploadCredsId, variable: 'kieUnpack']) >> mvnUploadCreds
-            1 * getPipelineMock('withCredentials')([mvnUploadCreds], _ as Closure)
-            1 * getPipelineMock('dir')(artifactDir, _ as Closure)
-            1 * getPipelineMock('sh')('zip -r artifacts .')
-            1 * getPipelineMock('sh')("curl --silent --upload-file artifacts.zip -u ${mvnUploadCreds} -v ${repoUrl}")
+        1 * getPipelineMock('usernameColonPassword.call')([credentialsId: mvnUploadCredsId, variable: 'kieUnpack']) >> mvnUploadCreds
+        1 * getPipelineMock('withCredentials')([mvnUploadCreds], _ as Closure)
+        1 * getPipelineMock('dir')(artifactDir, _ as Closure)
+        1 * getPipelineMock('sh')('zip -r artifacts .')
+        1 * getPipelineMock('sh')("curl --silent --upload-file artifacts.zip -u ${mvnUploadCreds} -v ${repoUrl}")
+    }
+
+    def "[maven.groovy] getLatestArtifactFromRepository OK"() {
```

Comment:
what about a test with nothing returned ?

Reply:
null test for both methods implemented